src/RestWrite.js

```diff
@@ -197,6 +200,37 @@ RestWrite.prototype.validateSchema = function () {
   );
 };
 
+// builds the default ACL depending on the className
+RestWrite.prototype.buildDefaultACL = function () {
+  if ((this.query && this.query.objectId) || this.data.ACL) {
+    return;
+  }
+  const defaultACL = {
+    currentUser: {
+      read: true,
+      write: true,
+    },
+  };
+  const aclOptions = Object.assign({}, this.config.defaultACL || defaultACL);
+  const reqUser = this.auth.user && this.auth.user.id;
+  if (reqUser && aclOptions.currentUser) {
+    aclOptions[reqUser] = aclOptions.currentUser;
+  }
```

Comment:
this code is not used at the right place.
You need to add an explicit test create/update ACL with the Rest API into spec/ParseACL.spec.js

Then the rest / graphql API could use this new field for better DX.

Here you only trigger the ACL transformation at creation time and if user do not provided ACL.

May be create a new restwrite fn `handleACL` to perform the operation after `buildDefaultACL`.

Reply:
I have added a test case for REST API and it works fine with `currentUser`